src/controller/java/src/chip/devicecontroller/model/InvokeElement.java

```diff
@@ -72,9 +111,30 @@ public String toString() {
         Locale.ENGLISH, "Endpoint %s, cluster %s, command %s", endpointId, clusterId, commandId);
   }
 
+  public static InvokeElement newInstance(
+      ChipPathId endpointId,
+      ChipPathId clusterId,
+      ChipPathId commandId,
```

Comment:
Not blocking this PR, but it seems we are repeating this pattern of a triple of endpoint / cluster / <target: attribute / command ID> all over for read/write/subscribe/invoke which ends up propagating into several overloads for raw integers vs object wrappers. Perhaps we could refactor this into a containing object with a builder that takes integer vs object for each property? Something like:

```
public final class InteractionTarget {
  private final ChipPathId endpointId;
  private final ChipPathId clusterId;
  private final ChipPathId targetId;

  public static final class Builder {
    private ChipPathId endpointId = ChipPathId.forWildcard();
    private ChipPathId clusterId = ChipPathId.forWildcard();
    private ChipPathId targetId = ChipPathId.forWildcard();
 
    public Builder setEndpointId(int endpointId) {
      this.endpointId = ChipPathId.forId(endpointId);
      return this;
    }

    public Builder setEndpointId(ChipPathId endpointId) {
      this.endpointId = endpointId;
      return this;
    }

    // Same for clusterId and targetId

    public InteractionTarget build() {
      return new InteractionTarget(endpointId, clusterId, targetId);
    }
  }
}
```

This way each of these `Element` classes can just take in an `InteractionTarget` and not have to require so many overloads.

Reply:
would refactor it in this ticket, https://github.com/project-chip/connectedhomeip/issues/25494, thanks for the suggestion.